src/main/java/taxi/dao/jdbcimpl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,210 @@
+import taxi.model.Manufacturer;
+import taxi.util.ConnectionUtil;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertQuery = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection
+                         .prepareStatement(insertQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject("GENERATED_KEY", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not add new manufacturer: "
+                    + car.toString(), e);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectQuery = "SELECT c.id AS car_id, model, "
+                + "m.manufacturer_name AS manufacturer_name,"
+                + "m.manufacturer_country AS manufacturer_country, "
+                + "m.id AS manufacturer_id "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.deleted = FALSE;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(createCar(resultSet, connection));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not find car with id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> carList = new ArrayList<>();
+        String getAllQuery = "SELECT c.id AS car_id, model, "
+                + "m.manufacturer_name AS manufacturer_name, "
+                + "m.manufacturer_country AS manufacturer_country, "
+                + "m.id AS manufacturer_id "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m ON m.id = c.manufacturer_id "
+                + "WHERE c.deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(createCar(resultSet, connection));
+            }
```

Comment:
same issues with this method - after the catch block call `getAllDriversByCarId` for each car in the carList
open a new connection in `getAllDriversByCarId` method

Reply:
Refactored as required.